src/StackAdmin/Azs.Backup.Admin/Module/Azs.Backup.Admin/Generated.PowerShell.Commands/SwaggerPathCommands/Set-AzsBackupShare.ps1

```diff
@@ -131,6 +150,16 @@ function Set-AzsBackupShare {
             $Location = $ArmResourceIdParameterValues['location']
         }
 
+        if ('Update' -eq $PsCmdlet.ParameterSetName -or 'ResourceId' -eq $PsCmdlet.ParameterSetName) {
+            if ($PSBoundParameters.ContainsKey('BackupShare') -or $PSBoundParameters.ContainsKey('Username') -or $PSBoundParameters.ContainsKey('Password'))
+            {
+                if (-not ($PSBoundParameters.ContainsKey('BackupShare') -and $PSBoundParameters.ContainsKey('Username') -and $PSBoundParameters.ContainsKey('Password')))
+                {
+                    throw "-BackupShare, -Username and -Password need to be all specified when trying to configure backup share path."
```

Comment:
In order to support Set-AzsBackupShare -BackupFrequencyInHours xxx, I have to either make all fields optional and add check logic inside; or add 2 more parameter sets (one for 'ResourceId' and another for 'Update').

We have a future feature to support Azure Storage as external store, and that will introduce parameters mutex against $BackupShare, then introducing 2 more parameter sets if we want to keep mandatory parameters. So I prefer we make all parameters optional and move check logic inside.

Reply:
BTW, I know it's not a good practice for this same endpoint, but is there any hard blocker if we just add two more cmdlets Set-AzsBackupScheduler and Set-AzsBackupRetentionPolicy instead of making change here? That will make everything easier.

Reply:
There is too much logic going on for something that should just be a REST call.  The client side should not be making any decisions like this, this should be a server side check.